clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -108,7 +133,7 @@ abstract class IndexService[CI <: ClioIndex](
   }
 
   def validateUpsert(
-    existingMetadata: Json,
+    existingMetadata: clioIndex.MetadataType,
     newMetadata: clioIndex.MetadataType
   ): Either[Exception, clioIndex.MetadataType] = {
     val differences = JsonDiff.diff(existingMetadata, newMetadata, remember = true)
```

Comment:
Woah, didn't notice this before... `JsonDiff.diff` works with case classes?

Reply:
yup!